simulation/network_simulator.cc

```diff
@@ -82,16 +103,14 @@ int main(int argc, char* argv[])
                 //auto raw = t1.count();
                 //esc.write(0x1800, &raw, sizeof(decltype(raw)));
                 esc.processDatagram(header, data, wkc);
+            }
 
-                mbx.receive();
-                mbx.process();
-                mbx.send();
-
-                esc.routine();
-
-                if (esc.al_status() & State::SAFE_OP)
+            for (auto& slave : slaves)
+            {
+                slave.routine();
+                if (slave.getState() == State::SAFE_OP)
                 {
-                    esc.set_valid_output_data_received(true);
+                    slave.setOutputDataValid(true);
```

Comment:
Never use with false: seems that "validateOutputData()" a batter naming and behavior.